Sources/TuistGenerator/Linter/StaticProductsGraphLinter.swift

```diff
@@ -52,72 +55,101 @@ class StaticProductsGraphLinter: StaticProductsGraphLinting {
     /// - In the event a node is a node capable of linking static products, it removes all the nodes
     ///   from the unlinked bucket and places them in the linked bucket in format of _staticNode > [linkingNode]_.
     ///
-    private func buildStaticProductsMap(visiting node: GraphNode,
+    private func buildStaticProductsMap(visiting dependency: ValueGraphDependency,
+                                        graphTraverser: GraphTraversing,
                                         cache: Cache) -> StaticProducts
     {
-        if let cachedResult = cache.results(for: node) {
+        if let cachedResult = cache.results(for: dependency) {
             return cachedResult
         }
 
         // Collect dependency results traversing the graph (dfs)
-        var results = dependencies(for: node).reduce(StaticProducts()) { results, node in
-            buildStaticProductsMap(visiting: node, cache: cache).merged(with: results)
+        var results = dependencies(for: dependency, graphTraverser: graphTraverser).reduce(StaticProducts()) { results, _ in
+            buildStaticProductsMap(visiting: dependency, graphTraverser: graphTraverser, cache: cache).merged(with: results)
```

Comment:
yikes this causes an infinite loop / stack overflow 😅 - missing using the `next` element from the reduce closure